src/treeViewLemma.ts

```diff
@@ -0,0 +1,129 @@
+      //   (entry) => entry.key.startsWith("1")
+      // );
+
+      // This is a temporary solution to return root entries
+      const rootEntries = new Array<Entry>();
+      rootEntries[0] = new Entry(
+        1,
+        "1H001",
+        "Hebrew",
+        vscode.TreeItemCollapsibleState.Collapsed
+      );
+      rootEntries[1] = new Entry(
+        1,
+        "1G002",
+        "Greek",
+        vscode.TreeItemCollapsibleState.Collapsed
+      );
+
+      return Promise.resolve(rootEntries);
+    } else {
+      // Generate parentKey for child entries
+      const key = element.key;
+      const parentKey =
+        (parseInt(element.level.toString()) + 1).toString() + key.substring(1);
+      const childKeyPrefix =
+        (element.level + 1).toString() + element.key.substring(1);
+      const childEntries = Array.from(Entry.entriesMap.values()).filter(
+        (entry) => entry.key.startsWith(childKeyPrefix)
+      );
+      return Promise.resolve(childEntries);
+    }
+  }
+}
+
+export class Entry extends vscode.TreeItem {
+  public static entriesMap: Map<string, Entry> = new Map<string, Entry>();
+  public static rootPath: string;
+  constructor(
+    public readonly level: number,
+    public readonly key: string,
+    public readonly label: string,
+    public readonly collapsibleState: vscode.TreeItemCollapsibleState,
+    public readonly command?: vscode.Command
+  ) {
+    super(label, collapsibleState);
+    this.tooltip = `${this.label}`;
+  }
+  public static async initialize(context: vscode.ExtensionContext) {
+    Entry.rootPath = context.extensionPath;
+    const fs = vscode.workspace.fs;
+    const hebrewDataPath = vscode.Uri.file(
+      path.join(Entry.rootPath, "data", "local", "treeViewHebrewWordData.json")
+    );
+    const greekDataPath = vscode.Uri.file(
+      path.join(Entry.rootPath, "data", "local", "treeViewGreekWordData.json")
+    );
+    console.log("hebrewDataPath:", hebrewDataPath);
+    console.log("greekDataPath:", greekDataPath);
+
+    await this.readAndProcessFile(hebrewDataPath, "hebrew");
+    await this.readAndProcessFile(greekDataPath, "greek");
+  }
+
+  private static async readAndProcessFile(
+    filePath: vscode.Uri,
+    language: string
+  ) {
+    try {
+      const fileContentUint8Array = await vscode.workspace.fs.readFile(
+        filePath
+      );
+      const fileContent = Buffer.from(fileContentUint8Array).toString("utf8");
+      const data = JSON.parse(fileContent);
+      for (const item of data) {
+        const collapsibleState =
+          item.level < 4
+            ? vscode.TreeItemCollapsibleState.Collapsed
+            : vscode.TreeItemCollapsibleState.None;
+        const command =
+          item.level === 4
+            ? {
+                command: "extension.openFile",
+                title: "Open File",
+                arguments: [
+                  `${Entry.rootPath}/data/eng/${language}/${item.fileName}${wordExtension}`,
```

Comment:
Two suggestions here:

1. This opens the file in the extension directory rather than the file in the open folder. We need to use `vscode.workspace.workspaceFolders` to get the Uri to the open workspace folder, and use that to build the Uri to the entry xml file
2. We can use the built in `vscode.open` command to open the file instead of making our own command. You can find it in [this list](https://code.visualstudio.com/api/references/commands)